### PR TITLE
Close MPD connection following an exception in handler.

### PR DESCRIPTION
**Pull request by kingosticks**
This goes towards fixing #1762. Needs some tests and changelog etc.

### CODE REVIEW

**Comment by kingosticks** [comment]
Oops, I meant to open this as a "draft" PR.

File: mopidy/internal/network.py [hunk 1]
```diff
@@ -425,6 +425,10 @@ def on_receive(self, message):
         if not self.prevent_timeout:
             self.connection.enable_timeout()
 
+    def on_failure(self, exception_type, exception_value, traceback):
+        """Ensure that cleanup when actor fails."""
```
Comment (jodal) [inline mopidy/internal/network.py]: Feel free to improve the wording here and in the next method.

**Comment by jodal** [comment]
I guess this could go into 2.2.x and thus should be rebased upon `release-2.2`.

File: mopidy/internal/network.py [hunk 2]
```diff
@@ -425,6 +425,10 @@ def on_receive(self, message):
         if not self.prevent_timeout:
             self.connection.enable_timeout()
 
+    def on_failure(self, exception_type, exception_value, traceback):
```
Comment (adamcik) [inline mopidy/internal/network.py]: Should we log something in here at debug and/or warning level?

Comment (kingosticks) [reply mopidy/internal/network.py]: The actual exception will have already been logged by the actor so I thought we didn't need anything more, just to clean up.

Comment (kingosticks) [reply mopidy/internal/network.py]: I can provide a snippet showing what is logged when this triggers if that's helpful.

Comment (adamcik) [reply mopidy/internal/network.py]: Nah, that's fine, just wanted to make sure it was already logged.

**Comment by kingosticks** [comment]
I couldn't really find a way to easily do the test I wanted with an exception raised in the handler running in a different actor, but I added something at least.

**Pull request merged**